agilent_infiniium_scope.py

```diff
@@ -72,319 +72,350 @@ def channel_infoNONEXISTANT(self, c, channel):
                 elif isinstance(mode, bool):
                     mode = int(mode)
                 elif isinstance(mode, int):
                     pass       
-            yield dev.write('ACQ:AVER '+str(mode))
-            resp = yield dev.query('ACQ:AVER?')
+            yield dev.write('ACQ:AVER {}'.format(mode))
+        resp = yield dev.query('ACQ:AVER?')
         returnValue(resp) 
         
-    @setting(119, navg = 'i', returns = ['i'])
-    def numavg(self, c, navg = None):
+    @setting(119, navg='i', returns=['i'])
+    def numavg(self, c, navg=None):
         """Get or set number of averages
         """
         dev = self.selectedDevice(c)
         if navg is None:
             resp = yield dev.query('ACQ:COUN?')
-        else:    
-            yield dev.write('ACQ:COUN %d' %navg)
+        else:
+            yield dev.write('ACQ:COUN {}'.format(navg))
             resp = yield dev.query('ACQ:COUN?')
         navg_out = int(resp)
-        returnValue(navg_out)         
+        returnValue(navg_out)
+
+    @setting(131, channel='i', level='v', returns='v{level}')
+    def trigger_at(self, c, channel, level=None):
+        """Get or set the trigger source and voltage for edge mode triggering.
 
-    @setting(131, channel = 'i', level = 'v', returns = 'v{level}')
-    def trigger_at(self, c, channel, level = None):
-        """Get or set the trigger source and the trigger voltage for edge mode triggering.
-        Channel must be one of 0 (AUX), 1, 2, 3, 4.
+        Channel must be one of 0 (AUX), 1, 2, 3, 4, or 5 (LINE).
+        If trigger source is 5 (LINE) the level will be ignored
         """
         dev = self.selectedDevice(c)
-        if channel==0:
+        if channel not in [0, 1, 2, 3, 4, 5]:
+            raise ValueError('Invalid trigger channel: {}'
+                             'Valid channels are [0, 1, 2, 3, 4, 5]'
+                             '0 for AUX, 5 for LINE'.format(channel))
+        if channel == 0:
             channel = 'AUX'
+        elif channel == 5:
+            channel = 'LINE'
         elif isinstance(channel, int):
-            #channel = 'CHAN%d' %channel
-            pass
-        if channel not in [0,1,2,3,4]:
-            raise Exception('Select valid trigger channel')
-        if level is None:
-            resp = yield dev.query('TRIG:LEV? CHAN%s' %channel)
+            channel = 'CHAN{}'.format(channel)
+        yield dev.write('TRIG:EDGE:SOUR {}'.format(channel))
+        if channel != 'LINE':
+            yield dev.write('TRIG:LEV {}, {}'.format(channel, level))
+            # set trigger level
+            resp = yield dev.query('TRIG:LEV? {}'.format(channel))
         else:
-            yield dev.write('TRIG:EDGE:SOUR CHAN%s' %channel) #set channel up for edge triggering  
-            yield dev.write('TRIG:LEV CHAN%s,%f' %(channel,level)) #set trigger level
-            resp = yield dev.query('TRIG:LEV? CHAN%s' %channel)
+            resp = 0.0
```

Comment:
I would change this condition around to put the special case of line triggering near the check:
```python
if channel == 'LINE':
    # Cannot set trigger level when triggering off line input.
    # See http://www.keysight.com/upload/cmc_upload/All/9000_series_prog_ref.pdf#page=914
    level = 0.0
else:
    # Set trigger level.
    yield dev.write('TRIG:LEV {}, {}'.format(channel, level))
    resp = yield dev.query('TRIG:LEV? {}'.format(channel))
    level = float(resp)
returnValue(level)
```